doc/man3/ASYNC_WAIT_CTX_new.pod

```diff
@@ -26,6 +28,14 @@ waiting for asynchronous jobs to complete
                                     size_t *numaddfds, OSSL_ASYNC_FD *delfd,
                                     size_t *numdelfds);
  int ASYNC_WAIT_CTX_clear_fd(ASYNC_WAIT_CTX *ctx, const void *key);
+ int ASYNC_WAIT_CTX_set_callback(ASYNC_WAIT_CTX *ctx,
+                                   ASYNC_callback_fn callback,
+                                   void *callback_arg);
+ int ASYNC_WAIT_CTX_get_callback(ASYNC_WAIT_CTX *ctx,
+                                  ASYNC_callback_fn *callback,
+                                  void **callback_arg);
+ int ASYNC_WAIT_CTX_set_status(ASYNC_WAIT_CTX *ctx, int status);
+ int ASYNC_WAIT_CTX_get_status(ASYNC_WAIT_CTX *ctx);
```

Comment:
My understanding of this is that an application would, in principle, have to be prepared to handle feedback via either the callback mechanism, or via fds - because it would not know whether the engine in question supports the callback. Similarly engines would have to be prepared to feedback via either mechanism because it would have to handle applications that don't register a callback. This seems less than ideal - but I can't think of a way around it.

Reply:
As we mentioned, this pull request is to provide an enhancement to the current OpenSSL async
notification mechanism, so if application wants to use this new method, our current solution is to call "ENGINE_ctrl_cmd" to check if engine support it or not.  In the long term, we can make it to be standard engine capability check. In the engine part, it can both method support by checking if callback is set. 
Since this method is more efficient, many people will like it, and engine code can add this support very quickly.